### PR TITLE
BREAKING(ext/net): improved error code accuracy

### LINKED ISSUE #13086: v2: change tcp/unix listener error types
Context: https://github.com/denoland/deno/pull/13075

Summary: change "Listener has been closed" `Deno.errors.BadResource` error to `Deno.errors.Interrupted`.

### PR DESCRIPTION
**Pull request by lucacasonato**
Fixes #13086

### CODE REVIEW

**Review by bartlomieju — APPROVED** [review]
LGTM!

File: ext/net/ops_unix.rs [hunk 1]
```diff
@@ -68,16 +67,18 @@ pub async fn op_net_accept_unix(
     .borrow()
     .resource_table
     .get::<NetworkListenerResource<UnixListener>>(rid)
-    .map_err(|_| bad_resource("Listener has been closed"))?;
+    .map_err(|_| {
+      std::io::Error::new(
+        std::io::ErrorKind::Interrupted,
```
Comment (piscisaureus) [inline ext/net/ops_unix.rs:72]: Interrupted is not really an appropriate error code in this case. Do you have something like Aborted ?

Comment (lucacasonato) [reply ext/net/ops_unix.rs:72]: You suggested to use `Interrupted` here: https://github.com/denoland/deno/pull/13075#issuecomment-992971370. We do not have an `Aborted` error.

I can introduce a new `Closed` error, and use that everywhere we throw a `BadResource` error - `BadResource` is not descriptive anyway.

Comment (bartlomieju) [reply ext/net/ops_unix.rs:72]: `Closed` seems reasonable - `BadResource` is a catch-all that is completely non-descriptive to users. Maybe we could make `Closed extends BadResource` to limit the impact as there are a lot of checks for `Deno.errors.BadResource`.

**Pull request merged**